specification/managednetworkfabric/resource-manager/Microsoft.ManagedNetworkFabric/preview/2024-02-15-preview/InternetGatewayRules.json

```diff
@@ -0,0 +1,510 @@
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "InternetGatewayRules_ListByResourceGroup_MaximumSet_Gen": {
+            "$ref": "./examples/InternetGatewayRules_ListByResourceGroup_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ManagedNetworkFabric/internetGatewayRules": {
+      "get": {
+        "operationId": "InternetGatewayRules_ListBySubscription",
+        "summary": "List Internet Gateway Rules by subscription.",
+        "description": "List all Internet Gateway rules in the given subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists all Internet Gateway Rules under the subscription.",
+            "schema": {
+              "$ref": "#/definitions/InternetGatewayRulesListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "InternetGatewayRules_ListBySubscription_MaximumSet_Gen": {
+            "$ref": "./examples/InternetGatewayRules_ListBySubscription_MaximumSet_Gen.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "InternetGatewayRuleProperties": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "required": [
+        "ruleProperties"
+      ],
+      "allOf": [
+        {
+          "$ref": "./common.json#/definitions/AnnotationResource",
+          "description": "Common properties for annotated type."
+        }
+      ],
+      "properties": {
+        "ruleProperties": {
+          "description": "Rules for the InternetGateways",
+          "$ref": "#/definitions/RuleProperties"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "./common.json#/definitions/ProvisioningState",
+          "readOnly": true
+        },
+        "internetGatewayIds": {
+          "type": "array",
+          "description": "List of Internet Gateway resource Id.",
+          "items": {
+            "description": "ARM Resource ID of the Internet Gateway.",
+            "type": "string"
```

Comment:
>type [](http://example.com/codeflow?start=13&length=4)

Please use "format": "arm-id" and "x-ms-arm-id-details" to specify the allowed resource types in this id array